packages/fxa-shared/payments/stripe.ts

```diff
@@ -0,0 +1,493 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import cacheManager, { Cacheable } from '@type-cacheable/core';
+import { useAdapter } from '@type-cacheable/ioredis-adapter';
+import {
+  deleteAccountCustomer,
+  getAccountCustomerByUid,
+} from '../db/models/auth';
+import { formatPlanConfigDto } from '../dto/auth/payments/plan-configuration';
+import { mapPlanConfigsByPriceId } from '../subscriptions/configuration/utils';
+
+import {
+  AbbrevPlan,
+  AbbrevPlayPurchase,
+  ConfiguredPlan,
+} from '../subscriptions/types';
+import { StatsD } from 'hot-shots';
+import ioredis from 'ioredis';
+import mapValues from 'lodash/mapValues';
+import { Stripe } from 'stripe';
+import { ILogger } from '../log';
+import { PaymentConfigManager } from './configuration/manager';
+import { FirestoreStripeError, StripeFirestore } from './stripe-firestore';
+
+export const CHARGES_RESOURCE = 'charges';
+export const COUPON_RESOURCE = 'coupons';
+export const CREDIT_NOTE_RESOURCE = 'creditNotes';
+export const CUSTOMER_RESOURCE = 'customers';
+export const INVOICES_RESOURCE = 'invoices';
+export const PAYMENT_METHOD_RESOURCE = 'paymentMethods';
+export const PLAN_RESOURCE = 'plans';
+export const PRICE_RESOURCE = 'prices';
+export const PRODUCT_RESOURCE = 'products';
+export const SOURCE_RESOURCE = 'sources';
+export const SUBSCRIPTIONS_RESOURCE = 'subscriptions';
+export const TAX_RATE_RESOURCE = 'taxRates';
+export const STRIPE_PLANS_CACHE_KEY = 'listStripePlans';
+export const STRIPE_PRODUCTS_CACHE_KEY = 'listStripeProducts';
+
+export const STRIPE_OBJECT_TYPE_TO_RESOURCE: Record<string, string> = {
+  charge: CHARGES_RESOURCE,
+  coupon: COUPON_RESOURCE,
+  credit_note: CREDIT_NOTE_RESOURCE,
+  customer: CUSTOMER_RESOURCE,
+  invoice: INVOICES_RESOURCE,
+  payment_method: PAYMENT_METHOD_RESOURCE,
+  plan: PLAN_RESOURCE,
+  price: PRICE_RESOURCE,
+  product: PRODUCT_RESOURCE,
+  source: SOURCE_RESOURCE,
+  subscription: SUBSCRIPTIONS_RESOURCE,
+  tax_rate: TAX_RATE_RESOURCE,
+};
+
+export const VALID_RESOURCE_TYPES = Object.values(
+  STRIPE_OBJECT_TYPE_TO_RESOURCE
+);
+
+export enum STRIPE_PRICE_METADATA {
+  PROMOTION_CODES = 'promotionCodes',
+  PLAY_SKU_IDS = 'playSkuIds',
+}
+
+export type StripeHelperConfig = {
+  subscriptions: {
+    stripeApiKey: string;
+    stripeWebhookSecret: string;
+    productConfigsFirestore: {
+      enabled: boolean;
+    };
+  };
+  authFirestore: {
+    prefix: string;
+  };
+  subhub: {
+    plansCacheTtlSeconds: number;
+  };
+  redis: any; // TODO
+};
```

Comment:
Ah cool, this is what I had thought would be around for the other refactored Base classes. 👍🏻